coalib/bearlib/aspects/meta.py

```diff
@@ -76,21 +77,41 @@ def __repr__(cls):
         return '<%s %s>' % (type(cls).__name__, repr(cls.__qualname__))
 
 
+def isaspect(item):
+    """
+    This function checks whether or not an object is an instance of
+    ``aspectclass``
+    """
+    if not isinstance(item, (aspectclass, aspectbase)):
+        return False
+
+    return True
+
+
+def assert_aspect(item):
+    """
+    This function raises ``aspectTypeError`` when an object is not an instance
+    of ``aspectclass``
+    """
+    class aspectTypeError(TypeError):
+
+        def __init__(self, message, severity=RESULT_SEVERITY.MAJOR):
+            super(aspectTypeError, self).__init__(message)
+            self.severity = severity
+
+    if not isaspect(item):
+        raise aspectTypeError(
+            '{} is not an aspectclass or an instance of an '
+            'aspectclass'.format(repr(item)))
+
```

Comment:
As already mentioned in my comment to your `aspectlist.__init__` changes: Just `return item` in the end, making the usage of `assert_aspect` even more powerful :)